// @seed/elements/es/avatar/vue/index.js
import { prefixCls } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';
import ImageCover from '../../image-cover/vue/index.js';
import 'prop-types';

//

var script = {
  name: 'avatar',
  components: {
    'image-cover': ImageCover,
  },

  props: {
    styles: {
      type: Object,
    },
    width: {
      type: Number,
    },
    url: {
      type: String,
      default: '',
    },
    loading: {
      type: Boolean,
      default: false,
    },
    imageCoverData: {
      type: Object,
    },
  },

  computed: {
    wrapStyle: function wrapStyle() {
      var style = {
        width: this.width / 2 + 'px',
        height: this.width / 2 + 'px',
        backgroundImage: this.url ? ("url(" + (this.url) + ")") : null,
      };
      return Object.assign({}, style, this.styles);
    },
    avatarClassnames: function avatarClassnames() {
      return [
        prefixCls('avatar-container'),
        this.loading ? prefixCls('avatar-loading') : '' ].join(' ');
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.avatarClassnames,style:(_vm.wrapStyle)},[(_vm.imageCoverData)?_c('image-cover',_vm._b({},'image-cover',_vm.imageCoverData,false)):_vm._e()],1)};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
