// @seed/elements/es/tag/weapp/services.js
/**
 * @since 2018-11-30 10:44
 * @author vivaxy
 */
export var BUY = 'buy';
export var GROUP_PURCHASE = 'group-purchase';
export var CASH_BACK = 'cash-back';
export var COUPON = 'coupon';
export var PAYMENT = 'payment';
export var TICKET = 'ticket';
export var REDUCE = 'reduce';
export var GIFT = 'gift';
export var TOUR = 'tour';
export var RENT = 'rent';
export var ACTIVITY = 'activity';
export var NEW = 'new';
export var SALE = 'sale';
export var SELL = 'sell';
export var BANK_CARD = 'bank-card';
export var PRICE = 'price';
export var VOUCHER = 'voucher';
export var PACKAGE = 'package';
export var SMART = 'smart';
export var BOOKING = 'booking';
export var TICKET2 = 'ticket2';
export var HOME = 'home';
export var ENDORSEMENT = 'endorsement';
export var SEAT = 'seat';
export var PHOTO = 'photo';
export var CERTIFICATE = 'certificate';
export var CARD = 'card';
export var NUMBER = 'number';
export var COURSE = 'course';
export var TAKEAWAY = 'takeaway';
export var ORDER = 'order';
export var QUEUE = 'queue';
export var PIECE = 'piece';
export var STYLE = 'style';
export var WEDDING_RECEPTION = 'wedding-reception';
export var SUNSHINE_RESTAURANT = 'sunshine-restaurant';
