// @seed/elements/lib/placeholder/vue/index.js
'use strict';

var uiBase = require('@seed/ui-base');
var normalizeComponent = require('../../normalize-component-136d5168.js');
require('../style/index.css');

//

var script = {
  name: 'placeholder',
  props: {
    type: {
      type: String,
      default: 'retrieval',
      validator: function validator(value) {
        return [
          'retrieval',
          'error',
          'address',
          'collection',
          'comment',
          'discount',
          'good-friend',
          'network',
          'news',
          'order',
          'search' ].includes(value);
      },
    },
  },
  computed: {
    classes: function classes() {
      return {
        placeholder: [
          uiBase.prefixCls('placeholder'),
          uiBase.prefixCls('placeholder', this.type) ],
      };
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.classes.placeholder})};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = normalizeComponent.__vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

module.exports = index;
