// @seed/elements/es/switch/vue/index.js
import { prefixCls } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//

var script = {
  name: 'switch',
  props: {
    value: null,
    checked: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    // 为了兼容 v-model 入参在 value 上的问题，并同时支持 checked 属性
    localChecked: function localChecked() {
      if (this.value !== undefined) {
        return !!this.value;
      }

      return this.checked;
    },
    className: function className() {
      var obj, obj$1;

      return [
        prefixCls('switch'),
        ( obj = {}, obj[prefixCls('switch--checked')] = this.localChecked, obj ),
        ( obj$1 = {}, obj$1[prefixCls('switch--disabled')] = this.disabled, obj$1 ) ];
    },
  },
  methods: {
    toggle: function toggle() {
      if (this.disabled) { return; }

      this.$emit('change', !this.localChecked);
      // 兼容 v-model 使用方式
      this.$emit('input', !this.localChecked);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('span',{class:_vm.className,on:{"click":_vm.toggle}})};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
