// @seed/elements/lib/separator/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _uiBase = require("@seed/ui-base");

require("../style/index.css");

/**
 * @since 2018-11-30 17:06
 * @author vivaxy
 */
var Separator =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Separator, _Component);

  function Separator() {
    (0, _classCallCheck2.default)(this, Separator);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Separator).apply(this, arguments));
  }

  (0, _createClass2.default)(Separator, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          color = _this$props.color,
          marginHorizontal = _this$props.marginHorizontal;
      return _react.default.createElement("div", {
        className: (0, _classnames.default)((0, _uiBase.prefixCls)('separator'), (0, _uiBase.prefixCls)('separator', 'line', size)),
        style: {
          backgroundColor: color,
          marginLeft: marginHorizontal,
          marginRight: marginHorizontal
        }
      });
    }
  }]);
  return Separator;
}(_react.Component);

exports.default = Separator;
Separator.propTypes = {
  size: _propTypes.default.string,
  color: _propTypes.default.string,
  marginHorizontal: _propTypes.default.string
};
Separator.defaultProps = {
  size: 'sm',
  color: null,
  marginHorizontal: '0'
};
