// @seed/elements/lib/image-cover/vue/index.js
'use strict';

var uiBase = require('@seed/ui-base');
var normalizeComponent = require('../../normalize-component-136d5168.js');
require('../style/index.css');

//

var script = {
  name: 'image-cover',
  data: function() {
    return {
      PRIORITY_ENUM: {
        HIGH: 'HIGH',
        LOW: 'LOW',
      },
    };
  },
  computed: {
    displayUrl: function displayUrl() {
      return this.url && this.url.indexOf('http') < 0
        ? ("https:" + (this.url))
        : this.url;
    },
    style: function style() {
      return {
        backgroundImage: 'url(' + this.displayUrl + ')',
      };
    },
    wrapperClasses: function wrapperClasses() {
      if (this.className) {
        return [uiBase.prefixCls('img-cover-wrap'), this.className].join(' ');
      } else {
        return [uiBase.prefixCls('img-cover-wrap')].join(' ');
      }
    },
  },
  props: {
    className: {
      type: String,
      default: '',
    },
    priority: {
      type: String,
      default: 'HIGH',
    },
    url: {
      type: String,
      default: '',
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.wrapperClasses},[(_vm.priority === _vm.PRIORITY_ENUM.HIGH)?_c('img',{staticClass:"seed-img-src",attrs:{"src":_vm.displayUrl,"alt":""}}):_vm._e(),_vm._v(" "),(_vm.priority === _vm.PRIORITY_ENUM.LOW)?_c('div',{staticClass:"seed-img-bg",style:(_vm.style)}):_vm._e()])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var ImageCover = normalizeComponent.__vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

module.exports = ImageCover;
