// @seed/elements/es/currency/react/index.js
import _classCallCheck from "@babel/runtime-corejs3/helpers/classCallCheck";
import _createClass from "@babel/runtime-corejs3/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime-corejs3/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime-corejs3/helpers/getPrototypeOf";
import _inherits from "@babel/runtime-corejs3/helpers/inherits";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cls from 'classnames';
import "../style/index.css";
import { prefixCls, priceZeroClear } from '@seed/ui-base';
var subTypeText = {
  none: '',
  start: '起',
  avg: '/人'
};

var Currency =
/*#__PURE__*/
function (_Component) {
  _inherits(Currency, _Component);

  function Currency() {
    _classCallCheck(this, Currency);

    return _possibleConstructorReturn(this, _getPrototypeOf(Currency).apply(this, arguments));
  }

  _createClass(Currency, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          clearZero = _this$props.clearZero,
          price = _this$props.price,
          discount = _this$props.discount,
          originPrice = _this$props.originPrice,
          subType = _this$props.subType,
          size = _this$props.size,
          styles = _this$props.styles;
      var wrapclasses = cls(prefixCls('price'), prefixCls(size ? "price_".concat(size) : 'price_m'));

      if (size == 's') {
        return React.createElement("div", {
          style: styles,
          className: wrapclasses
        }, price ? '￥' + priceZeroClear(clearZero, price) : '暂无售价', subType ? subTypeText[subType] : null);
      }

      return React.createElement("div", {
        style: styles,
        className: prefixCls('price')
      }, React.createElement("div", {
        className: prefixCls(size ? "price_".concat(size) : 'price_m')
      }, React.createElement("i", null, "\uFFE5"), price ? priceZeroClear(clearZero, price) : '暂无售价'), subType && !originPrice ? React.createElement("span", null, subTypeText[subType]) : null, React.createElement("div", {
        className: prefixCls('origin_price_wrap')
      }, discount ? React.createElement("div", {
        className: prefixCls('discount_tag')
      }, discount) : null, originPrice ? React.createElement("span", null, "\uFFE5", React.createElement("del", null, priceZeroClear(clearZero, originPrice))) : null));
    }
  }]);

  return Currency;
}(Component);

export { Currency as default };
Currency.propTypes = {
  price: PropTypes.number,
  originPrice: PropTypes.number,
  clearZero: PropTypes.bool,
  //是否要移除价格末尾为0的部分,默认为true
  subType: PropTypes.string,
  //附加信息 起、人均
  discount: PropTypes.string,
  //折扣标签
  size: PropTypes.oneOf(['s', 'm', 'l']),
  styles: PropTypes.object
};
Currency.defaultProps = {
  size: 's',
  originPrice: 0,
  price: 0,
  subType: 'none',
  clearZero: true
};
