// @seed/elements/es/button/vue/index.js
import { prefixCls } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//
var script = {
  name: 'button',
  props: {
    type: {
      type: Number,
      default: 1,
    },
    level: {
      type: Number,
      default: 1,
    },
    state: {
      type: String,
      default: '', //按钮的四种状态，
    },
    size: {
      type: String,
      validator: function(value) {
        return ['l', 's', 'm'].indexOf(value) > -1;
      },
    },
    icon: {
      type: Element,
      default: null,
    },
    styles: {
      type: Object,
    },
    text: {
      type: String,
    },
  },
  data: function data() {
    return {
      press: false,
    };
  },
  computed: {
    classes: function classes() {
      return [
        prefixCls("btn"),
        prefixCls(this.getBtnStyle),
        this.size ? prefixCls(("size_" + (this.size))) : prefixCls('size_m') ];
    },
    type2classes: function type2classes() {
      var obj;

      return [
        prefixCls('btn', 'type2'),
        prefixCls('btn', 'type2', ("sub" + (this.subType))),
        ( obj = {}, obj[prefixCls('bold')] = this.bold, obj ) ];
    },
    type3classes: function type3classes() {
      var obj;

      return [
        prefixCls('btn', 'type3'),
        prefixCls('btn', 'type3', ("sub" + (this.subType))),
        ( obj = {}, obj[prefixCls('bold')] = this.bold, obj ) ];
    },
    getBtnStyle: function getBtnStyle() {
      // debugger
      var ref = this;
      var press = ref.press;
      var level = ref.level;
      var state = press ? 'press' : this.state;
      // 状态、层级都为空
      if (!state && level) {
        return ("btn" + level);
      }
      if (state) {
        var lv = level ? level : 1;
        switch (state) {
          case 'normal':
            return ("btn" + lv);
          case 'press':
            return ("btn" + lv + "_press");
          case 'awaiting':
            return ("btn" + lv + "_await");
          case 'disable':
            return "btn_disable";
        }
      }
      return "btn1";
    },
  },
  methods: {
    handleClick: function handleClick(e) {
      this.$emit('click', e, this);
    },
    touchStart: function touchStart(e) {
      if (this.state == 'disable') { return; }
      this.press = true;
      // this.$emit('click', e, this);
    },
    touchEnd: function touchEnd(e) {
      if (this.state == 'disable') { return; }
      this.press = false;
      this.$emit('click', e, this);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return (_vm.type === 2)?_c('a',{class:_vm.type2classes,style:(_vm.style),on:{"click":_vm.handleClick}},[_vm._t("default")],2):(_vm.type === 3)?_c('div',{class:_vm.type3classes,style:(_vm.style),on:{"click":_vm.handleClick}},[_vm._t("default")],2):(_vm.type === 1)?_c('div',{class:_vm.classes,on:{"touchstart":_vm.touchStart,"touchend":_vm.touchEnd}},[_vm._t("default")],2):_vm._e()};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
