// @seed/elements/es/loading/react/index.js
import _defineProperty from "@babel/runtime-corejs3/helpers/defineProperty";
import _classCallCheck from "@babel/runtime-corejs3/helpers/classCallCheck";
import _createClass from "@babel/runtime-corejs3/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime-corejs3/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime-corejs3/helpers/getPrototypeOf";
import _inherits from "@babel/runtime-corejs3/helpers/inherits";
import React from 'react';
import PropTypes from 'prop-types';
import "../style/index.css";
import cls from 'classnames';
import { prefixCls } from '@seed/ui-base';

var Loading =
/*#__PURE__*/
function (_React$Component) {
  _inherits(Loading, _React$Component);

  function Loading(props) {
    _classCallCheck(this, Loading);

    return _possibleConstructorReturn(this, _getPrototypeOf(Loading).call(this, props));
  }

  _createClass(Loading, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          show = _this$props.show,
          loadingText = _this$props.loadingText,
          loadingStyles = _this$props.loadingStyles,
          vCenter = _this$props.vCenter;

      if (show === false) {
        return null;
      }

      return React.createElement("div", {
        style: loadingStyles,
        className: cls(prefixCls('loading-container'), _defineProperty({}, prefixCls('v-center'), vCenter))
      }, React.createElement("div", {
        className: prefixCls('load-wrap')
      }, React.createElement("div", {
        className: prefixCls('loading-img')
      }), React.createElement("span", null, loadingText ? loadingText : '')));
    }
  }]);

  return Loading;
}(React.Component);

export { Loading as default };
Loading.propTypes = {
  loading: PropTypes.bool,
  // 增加icon
  // 增加overlay
  vCenter: PropTypes.bool,
  loadingStyles: PropTypes.object,
  loadingText: PropTypes.string
};
