// @seed/elements/es/page-control/vue/index.js
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//
var script = {
  name: 'page-control',
  props: {
    size: {
      type: String,
      validator: function(value) {
        return ['', 'lg', 'sm'].indexOf(value) > -1;
      },
    },
    selectedColor: String,
    color: String,
    count: Number,
    selectedIndex: Number,
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('ul',{class:{
    'seed-page-control': true,
    'seed-page-control--sm': _vm.size === 'sm',
  }},_vm._l((_vm.count),function(index){return _c('li',{key:index,class:{
      'seed-page-control__item': true,
      'seed-page-control__item--sm': _vm.size === 'sm',
      'seed-page-control__item--selected': index - 1 == _vm.selectedIndex,
    },style:({
      backgroundColor: index - 1 == _vm.selectedIndex ? _vm.selectedColor : _vm.color,
    })})}),0)};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
