// @seed/elements/src/currency/react/index.js
import React, { Component } from 'react';

import PropTypes from 'prop-types';
import cls from 'classnames';
import '../style/index.less';
import { prefixCls, priceZeroClear } from '@seed/ui-base';

const subTypeText = {
  none: '',
  start: '起',
  avg: '/人',
};

export default class Currency extends Component {
  render() {
    const {
      clearZero,
      price,
      discount,
      originPrice,
      subType,
      size,
      styles,
    } = this.props;
    const wrapclasses = cls(
      prefixCls('price'),
      prefixCls(size ? `price_${size}` : 'price_m'),
    );
    if (size == 's') {
      return (
        <div style={styles} className={wrapclasses}>
          {price ? '￥' + priceZeroClear(clearZero, price) : '暂无售价'}
          {subType ? subTypeText[subType] : null}
        </div>
      );
    }
    return (
      <div style={styles} className={prefixCls('price')}>
        <div className={prefixCls(size ? `price_${size}` : 'price_m')}>
          <i>￥</i>
          {price ? priceZeroClear(clearZero, price) : '暂无售价'}
        </div>
        {subType && !originPrice ? <span>{subTypeText[subType]}</span> : null}
        <div className={prefixCls('origin_price_wrap')}>
          {discount ? (
            <div className={prefixCls('discount_tag')}>{discount}</div>
          ) : null}
          {originPrice ? (
            <span>
              ￥<del>{priceZeroClear(clearZero, originPrice)}</del>
            </span>
          ) : null}
        </div>
      </div>
    );
  }
}

Currency.propTypes = {
  price: PropTypes.number,
  originPrice: PropTypes.number,
  clearZero: PropTypes.bool, //是否要移除价格末尾为0的部分,默认为true
  subType: PropTypes.string, //附加信息 起、人均
  discount: PropTypes.string, //折扣标签
  size: PropTypes.oneOf(['s', 'm', 'l']),
  styles: PropTypes.object,
};
Currency.defaultProps = {
  size: 's',
  originPrice: 0,
  price: 0,
  subType: 'none',
  clearZero: true,
};
