// @seed/elements/lib/image-cover/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = ImageCover;
exports.PRIORITY_ENUM = void 0;

var _values = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/values"));

var _indexOf = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/index-of"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("../style/index.css");

var PRIORITY_ENUM = {
  HIGH: 'HIGH',
  LOW: 'LOW'
};
exports.PRIORITY_ENUM = PRIORITY_ENUM;

function ImageCover(props) {
  var url = props.url,
      className = props.className,
      priority = props.priority;
  var displayUrl = url;

  if (url && (0, _indexOf.default)(url).call(url, 'http') < 0) {
    displayUrl = "https:".concat(url);
  }

  return _react.default.createElement("div", {
    className: "seed-img-cover-wrap ".concat(className)
  }, priority === PRIORITY_ENUM.HIGH && _react.default.createElement("img", {
    className: "seed-img-src",
    src: displayUrl,
    alt: ""
  }), priority === PRIORITY_ENUM.LOW && _react.default.createElement("div", {
    className: "seed-img-bg",
    style: {
      backgroundImage: "url(".concat(displayUrl, ")")
    }
  }));
}

ImageCover.propTypes = {
  url: _propTypes.default.string,
  priority: _propTypes.default.oneOf((0, _values.default)(PRIORITY_ENUM)),
  className: _propTypes.default.string
};
ImageCover.defaultProps = {
  url: '',
  priority: PRIORITY_ENUM.HIGH,
  className: ''
};
