// @seed/elements/es/mtstar/vue/index.js
import { prefixCls } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/mtstar.css';

//

var STARS = Array.from({ length: 5 });

var script = {
  name: 'mtstar',
  props: {
    star: {
      type: Number,
      default: 0,
    },
  },
  computed: {
    starInfo: function starInfo() {
      var ref = this;
      var star = ref.star;
      var starValueRound = Math.round(star / 10);
      var starValueCeil = Math.ceil(star / 10);
      return STARS.map(function (item, index) { return ({
        value: index + 1 <= starValueRound ? index + 1 : 0,
        isHalf: star % 10 > 0 && index + 1 === starValueCeil,
      }); });
    },
    stars: function stars() {
      return STARS;
    },
  },
  methods: {
    prefixCls: function prefixCls$1() {
      var arg = [], len = arguments.length;
      while ( len-- ) arg[ len ] = arguments[ len ];

      return prefixCls.apply(void 0, arg);
    },
    starClass: function starClass(index) {
      return [
        prefixCls('star-mt'),
        prefixCls('star-mt-' + this.starInfo[index].value) ];
    },
    halfStar: function halfStar(index) {
      return this.starInfo[index].isHalf;
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.prefixCls('star-mt-wrap')},_vm._l((_vm.stars),function(item,index){return _c('div',{key:index,class:_vm.starClass(index)},[(_vm.halfStar(index))?_c('div',{class:_vm.prefixCls('star-mt-half')}):_vm._e()])}),0)};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
