// @seed/elements/src/page-control/react/index.js
import React, { Component } from 'react';

import PropTypes from 'prop-types';
import cls from 'classnames';
import '../style/index.less';

export default class PageControl extends Component {
  render() {
    const { size, selectedColor, color, count, selectedIndex } = this.props;
    const navWrapperCls = cls({
      'seed-page-control': true,
      'seed-page-control--sm': size === 'sm',
    });
    const navItemCls = cls({
      'seed-page-control__item': true,
      'seed-page-control__item--sm': size === 'sm',
    });
    return (
      <ul className={navWrapperCls}>
        {Array.from(Array(count), (_, index) => (
          <li
            className={cls(navItemCls, {
              'seed-page-control__item--selected': index == selectedIndex,
            })}
            style={{
              backgroundColor: index == selectedIndex ? selectedColor : color,
            }}
          ></li>
        ))}
      </ul>
    );
  }
}

PageControl.propTypes = {
  size: PropTypes.oneOf(['lg', 'sm']),
  selectedColor: PropTypes.string,
  color: PropTypes.string,
  count: PropTypes.number,
  selectedIndex: PropTypes.number,
};
