// @seed/elements/es/switch/react/index.js
import _defineProperty from "@babel/runtime-corejs3/helpers/defineProperty";
import _concatInstanceProperty from "@babel/runtime-corejs3/core-js-stable/instance/concat";
import _classCallCheck from "@babel/runtime-corejs3/helpers/classCallCheck";
import _createClass from "@babel/runtime-corejs3/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime-corejs3/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime-corejs3/helpers/getPrototypeOf";
import _inherits from "@babel/runtime-corejs3/helpers/inherits";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cls from 'classnames';
import { prefixCls } from '@seed/ui-base';
import "../style/index.css";

var Switch =
/*#__PURE__*/
function (_Component) {
  _inherits(Switch, _Component);

  function Switch() {
    var _getPrototypeOf2, _context;

    var _this;

    _classCallCheck(this, Switch);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Switch)).call.apply(_getPrototypeOf2, _concatInstanceProperty(_context = [this]).call(_context, args)));

    _this.toggle = function () {
      if (_this.props.disabled) {
        return;
      }

      var checked = !_this.props.checked;

      _this.props.onChange(checked);
    };

    return _this;
  }

  _createClass(Switch, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          checked = _this$props.checked,
          disabled = _this$props.disabled;
      return React.createElement("span", {
        className: cls(prefixCls('switch'), _defineProperty({}, prefixCls('switch--checked'), checked), _defineProperty({}, prefixCls('switch--disabled'), disabled)),
        onClick: this.toggle
      });
    }
  }]);

  return Switch;
}(Component);

Switch.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func
};
Switch.defaultProps = {
  checked: false,
  disabled: false,
  onChange: function onChange() {}
};
export { Switch as default };
