// @seed/elements/lib/page-control/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _from = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/array/from"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

require("../style/index.css");

var PageControl =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(PageControl, _Component);

  function PageControl() {
    (0, _classCallCheck2.default)(this, PageControl);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(PageControl).apply(this, arguments));
  }

  (0, _createClass2.default)(PageControl, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          selectedColor = _this$props.selectedColor,
          color = _this$props.color,
          count = _this$props.count,
          selectedIndex = _this$props.selectedIndex;
      var navWrapperCls = (0, _classnames.default)({
        'seed-page-control': true,
        'seed-page-control--sm': size === 'sm'
      });
      var navItemCls = (0, _classnames.default)({
        'seed-page-control__item': true,
        'seed-page-control__item--sm': size === 'sm'
      });
      return _react.default.createElement("ul", {
        className: navWrapperCls
      }, (0, _from.default)(Array(count), function (_, index) {
        return _react.default.createElement("li", {
          className: (0, _classnames.default)(navItemCls, {
            'seed-page-control__item--selected': index == selectedIndex
          }),
          style: {
            backgroundColor: index == selectedIndex ? selectedColor : color
          }
        });
      }));
    }
  }]);
  return PageControl;
}(_react.Component);

exports.default = PageControl;
PageControl.propTypes = {
  size: _propTypes.default.oneOf(['lg', 'sm']),
  selectedColor: _propTypes.default.string,
  color: _propTypes.default.string,
  count: _propTypes.default.number,
  selectedIndex: _propTypes.default.number
};
