// @seed/elements/es/avatar/react/index.js
import _Object$assign from "@babel/runtime-corejs3/core-js-stable/object/assign";
import React from 'react';
import PropTypes from 'prop-types';
import ImageCover from '../../image-cover/react/index';
import "../style/index.css";
import cls from 'classnames';
import { prefixCls } from '@seed/ui-base';
export default function Avatar(props) {
  var styles = props.styles,
      width = props.width,
      url = props.url,
      loading = props.loading,
      imageCoverData = props.imageCoverData;
  var wrapStyle = {
    width: width / 2 + 'px',
    height: width / 2 + 'px',
    backgroundImage: url ? "url(".concat(url, ")") : null
  };
  return React.createElement("div", {
    style: _Object$assign({}, wrapStyle, styles),
    className: cls(prefixCls('avatar-container'), loading ? prefixCls('avatar-loading') : '')
  }, imageCoverData && React.createElement(ImageCover, imageCoverData));
}
Avatar.propTypes = {
  // 增加icon
  // 增加overlay
  url: PropTypes.string,
  styles: PropTypes.object,
  width: PropTypes.number,
  defaultImg: PropTypes.bool,
  imageCoverData: PropTypes.object
};
