// @seed/elements/lib/currency/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

require("../style/index.css");

var _uiBase = require("@seed/ui-base");

var subTypeText = {
  none: '',
  start: '起',
  avg: '/人'
};

var Currency =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Currency, _Component);

  function Currency() {
    (0, _classCallCheck2.default)(this, Currency);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Currency).apply(this, arguments));
  }

  (0, _createClass2.default)(Currency, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          clearZero = _this$props.clearZero,
          price = _this$props.price,
          discount = _this$props.discount,
          originPrice = _this$props.originPrice,
          subType = _this$props.subType,
          size = _this$props.size,
          styles = _this$props.styles;
      var wrapclasses = (0, _classnames.default)((0, _uiBase.prefixCls)('price'), (0, _uiBase.prefixCls)(size ? "price_".concat(size) : 'price_m'));

      if (size == 's') {
        return _react.default.createElement("div", {
          style: styles,
          className: wrapclasses
        }, price ? '￥' + (0, _uiBase.priceZeroClear)(clearZero, price) : '暂无售价', subType ? subTypeText[subType] : null);
      }

      return _react.default.createElement("div", {
        style: styles,
        className: (0, _uiBase.prefixCls)('price')
      }, _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)(size ? "price_".concat(size) : 'price_m')
      }, _react.default.createElement("i", null, "\uFFE5"), price ? (0, _uiBase.priceZeroClear)(clearZero, price) : '暂无售价'), subType && !originPrice ? _react.default.createElement("span", null, subTypeText[subType]) : null, _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)('origin_price_wrap')
      }, discount ? _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)('discount_tag')
      }, discount) : null, originPrice ? _react.default.createElement("span", null, "\uFFE5", _react.default.createElement("del", null, (0, _uiBase.priceZeroClear)(clearZero, originPrice))) : null));
    }
  }]);
  return Currency;
}(_react.Component);

exports.default = Currency;
Currency.propTypes = {
  price: _propTypes.default.number,
  originPrice: _propTypes.default.number,
  clearZero: _propTypes.default.bool,
  //是否要移除价格末尾为0的部分,默认为true
  subType: _propTypes.default.string,
  //附加信息 起、人均
  discount: _propTypes.default.string,
  //折扣标签
  size: _propTypes.default.oneOf(['s', 'm', 'l']),
  styles: _propTypes.default.object
};
Currency.defaultProps = {
  size: 's',
  originPrice: 0,
  price: 0,
  subType: 'none',
  clearZero: true
};
