// @seed/elements/lib/checkbox/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _bind = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/bind"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("../style/index.css");

var Checkbox =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Checkbox, _Component);

  function Checkbox(props) {
    var _context;

    var _this;

    (0, _classCallCheck2.default)(this, Checkbox);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Checkbox).call(this, props));
    _this.state = {
      checked: props.checked
    };
    _this.onChange = (0, _bind.default)(_context = _this.onChange).call(_context, (0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(Checkbox, [{
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      var checked = nextProps.checked,
          onChange = nextProps.onChange;

      if (nextProps.checked !== this.state.checked) {
        this.setState({
          checked: checked
        });
        onChange && onChange(checked);
      }
    }
  }, {
    key: "onChange",
    value: function onChange() {
      var _this$props = this.props,
          disabled = _this$props.disabled,
          onChange = _this$props.onChange;
      if (disabled) return;
      var checked = !this.state.checked;
      this.setState({
        checked: checked
      });
      onChange && onChange(checked);
    }
  }, {
    key: "render",
    value: function render() {
      var disabled = this.props.disabled;
      return _react.default.createElement("label", {
        className: "seed-checkbox"
      }, _react.default.createElement("input", {
        type: "checkbox",
        className: "seed-checkbox__input",
        checked: this.state.checked,
        disabled: disabled,
        onChange: this.onChange
      }), _react.default.createElement("div", {
        className: "seed-checkbox__mark"
      }));
    }
  }]);
  return Checkbox;
}(_react.Component);

exports.default = Checkbox;
Checkbox.propTypes = {
  checked: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
