// @seed/elements/lib/star/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _map = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/map"));

var _bind = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/bind"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _utils = require("../util/utils");

var _uiBase = require("@seed/ui-base");

require("../style/index.css");

var Star =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Star, _React$Component);

  function Star(props) {
    var _context;

    var _this;

    (0, _classCallCheck2.default)(this, Star);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Star).call(this, props));
    _this.onClick = (0, _bind.default)(_context = _this.onClick).call(_context, (0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(Star, [{
    key: "onClick",
    value: function onClick(e) {
      var canClick = this.props.canClick;
      var index = e.target.dataset.index;

      if (canClick && index) {
        var _index = Number(index);

        this.props.onChange(_index);
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          star = _this$props.star,
          score = _this$props.score,
          starStyle = _this$props.starStyle,
          textStyle = _this$props.textStyle,
          classname = _this$props.classname;

      if (star < 0 || star === null) {
        return null;
      }

      var _star = (0, _utils.handleScore)(star) && (0, _utils.handleScore)(star).toString().replace('.', '_') || 0;

      return _react.default.createElement("div", {
        className: (0, _classnames.default)((0, _uiBase.prefixCls)('star_wrap'), classname)
      }, _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)('star_icon'),
        onClick: this.onClick
      }, (0, _map.default)(_utils.STAR).call(_utils.STAR, function (item, index) {
        return _react.default.createElement("span", {
          key: index,
          "data-index": item,
          className: (0, _classnames.default)((0, _uiBase.prefixCls)('star'), [(0, _uiBase.prefixCls)("star_".concat(_star))]),
          style: starStyle
        });
      })), score ? _react.default.createElement("div", {
        className: (0, _classnames.default)((0, _uiBase.prefixCls)('star_score'), [(0, _uiBase.prefixCls)("score_".concat(_star))]),
        style: textStyle
      }, score) : null);
    }
  }]);
  return Star;
}(_react.default.Component);

exports.default = Star;
Star.propTypes = {
  star: _propTypes.default.number,
  score: _propTypes.default.oneOfType([_propTypes.default.string, _propTypes.default.number]),
  starStyle: _propTypes.default.object,
  textStyle: _propTypes.default.object,
  classname: _propTypes.default.string,
  canClick: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
Star.defaultProps = {
  star: 0,
  score: '',
  starStyle: {},
  textStyle: {},
  classname: '',
  canClick: false,
  onChange: function onChange() {}
};
