// @seed/elements/lib/tag/common/services.js
"use strict";

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.SUNSHINE_RESTAURANT = exports.WEDDING_RECEPTION = exports.STYLE = exports.PIECE = exports.QUEUE = exports.ORDER = exports.TAKEAWAY = exports.COURSE = exports.NUMBER = exports.CARD = exports.CERTIFICATE = exports.PHOTO = exports.SEAT = exports.ENDORSEMENT = exports.HOME = exports.TICKET2 = exports.BOOKING = exports.SMART = exports.PACKAGE = exports.VOUCHER = exports.PRICE = exports.BANK_CARD = exports.SELL = exports.SALE = exports.NEW = exports.ACTIVITY = exports.RENT = exports.TOUR = exports.GIFT = exports.REDUCE = exports.TICKET = exports.PAYMENT = exports.COUPON = exports.CASH_BACK = exports.GROUP_PURCHASE = exports.BUY = void 0;

/**
 * @since 2018-11-30 10:44
 * @author vivaxy
 */
var BUY = 'buy';
exports.BUY = BUY;
var GROUP_PURCHASE = 'group-purchase';
exports.GROUP_PURCHASE = GROUP_PURCHASE;
var CASH_BACK = 'cash-back';
exports.CASH_BACK = CASH_BACK;
var COUPON = 'coupon';
exports.COUPON = COUPON;
var PAYMENT = 'payment';
exports.PAYMENT = PAYMENT;
var TICKET = 'ticket';
exports.TICKET = TICKET;
var REDUCE = 'reduce';
exports.REDUCE = REDUCE;
var GIFT = 'gift';
exports.GIFT = GIFT;
var TOUR = 'tour';
exports.TOUR = TOUR;
var RENT = 'rent';
exports.RENT = RENT;
var ACTIVITY = 'activity';
exports.ACTIVITY = ACTIVITY;
var NEW = 'new';
exports.NEW = NEW;
var SALE = 'sale';
exports.SALE = SALE;
var SELL = 'sell';
exports.SELL = SELL;
var BANK_CARD = 'bank-card';
exports.BANK_CARD = BANK_CARD;
var PRICE = 'price';
exports.PRICE = PRICE;
var VOUCHER = 'voucher';
exports.VOUCHER = VOUCHER;
var PACKAGE = 'package';
exports.PACKAGE = PACKAGE;
var SMART = 'smart';
exports.SMART = SMART;
var BOOKING = 'booking';
exports.BOOKING = BOOKING;
var TICKET2 = 'ticket2';
exports.TICKET2 = TICKET2;
var HOME = 'home';
exports.HOME = HOME;
var ENDORSEMENT = 'endorsement';
exports.ENDORSEMENT = ENDORSEMENT;
var SEAT = 'seat';
exports.SEAT = SEAT;
var PHOTO = 'photo';
exports.PHOTO = PHOTO;
var CERTIFICATE = 'certificate';
exports.CERTIFICATE = CERTIFICATE;
var CARD = 'card';
exports.CARD = CARD;
var NUMBER = 'number';
exports.NUMBER = NUMBER;
var COURSE = 'course';
exports.COURSE = COURSE;
var TAKEAWAY = 'takeaway';
exports.TAKEAWAY = TAKEAWAY;
var ORDER = 'order';
exports.ORDER = ORDER;
var QUEUE = 'queue';
exports.QUEUE = QUEUE;
var PIECE = 'piece';
exports.PIECE = PIECE;
var STYLE = 'style';
exports.STYLE = STYLE;
var WEDDING_RECEPTION = 'wedding-reception';
exports.WEDDING_RECEPTION = WEDDING_RECEPTION;
var SUNSHINE_RESTAURANT = 'sunshine-restaurant';
exports.SUNSHINE_RESTAURANT = SUNSHINE_RESTAURANT;
