// @seed/elements/lib/tag/common/types.js
"use strict";

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.VIP = exports.LEVEL = exports.TOP_QUALITY_REVIEW = exports.TOPIC = exports.FLAT_BLACK = exports.GHOST = exports.SECONDARY = exports.FLAT = exports.DISABLED = exports.PRIMARY = exports.MUST_GO_LIST = exports.MUST_LIVE_LIST = exports.MUST_STROLL_LIST = exports.MUST_PLAY_LIST = exports.MUST_EAT_LIST = exports.BLACK_PEARL = exports.CHAIN_CERTIFICATION = exports.RANKING_VERTICAL = exports.RANKING_HORIZONTAL = exports.SERVICE = void 0;

/**
 * @since 20181129 15:09
 * @author vivaxy
 */
var SERVICE = 'service';
exports.SERVICE = SERVICE;
var RANKING_HORIZONTAL = 'ranking-horizontal';
exports.RANKING_HORIZONTAL = RANKING_HORIZONTAL;
var RANKING_VERTICAL = 'ranking-vertical';
exports.RANKING_VERTICAL = RANKING_VERTICAL;
var CHAIN_CERTIFICATION = 'chain-certification';
exports.CHAIN_CERTIFICATION = CHAIN_CERTIFICATION;
var BLACK_PEARL = 'black-pearl';
exports.BLACK_PEARL = BLACK_PEARL;
var MUST_EAT_LIST = 'must-eat-list';
exports.MUST_EAT_LIST = MUST_EAT_LIST;
var MUST_PLAY_LIST = 'must-play-list';
exports.MUST_PLAY_LIST = MUST_PLAY_LIST;
var MUST_STROLL_LIST = 'must-stroll-list';
exports.MUST_STROLL_LIST = MUST_STROLL_LIST;
var MUST_LIVE_LIST = 'must-live-list';
exports.MUST_LIVE_LIST = MUST_LIVE_LIST;
var MUST_GO_LIST = 'must-go-list';
exports.MUST_GO_LIST = MUST_GO_LIST;
var PRIMARY = 'primary';
exports.PRIMARY = PRIMARY;
var DISABLED = 'disabled';
exports.DISABLED = DISABLED;
var FLAT = 'flat';
exports.FLAT = FLAT;
var SECONDARY = 'secondary';
exports.SECONDARY = SECONDARY;
var GHOST = 'ghost';
exports.GHOST = GHOST;
var FLAT_BLACK = 'flat-black';
exports.FLAT_BLACK = FLAT_BLACK;
var TOPIC = 'topic';
exports.TOPIC = TOPIC;
var TOP_QUALITY_REVIEW = 'top-quality-review';
exports.TOP_QUALITY_REVIEW = TOP_QUALITY_REVIEW;
var LEVEL = 'level';
exports.LEVEL = LEVEL;
var VIP = 'vip';
exports.VIP = VIP;
