// @seed/elements/es/animated-interaction/vue/index.js
import { prefixCls } from '@seed/ui-base';
import '../style/srcoll-num.css';
import ScrollNum from './scroll-num.js';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//
var script = {
  name: 'animated-interaction',
  data: function data() {
    return {
      animate: false,
    };
  },
  components: {
    'scroll-num': ScrollNum,
  },
  computed: {
    aniBoxClasses: function aniBoxClasses() {
      return ((prefixCls('ani-box')) + " " + (prefixCls(this.iconType)) + " " + (this.active ? prefixCls('active') : ''));
    },
    iconClasses: function iconClasses() {
      return ((prefixCls(this.iconType + '-icon' + '-' + this.theme)) + " " + (this.animate ? prefixCls('animate ') : ''));
    },
    iconDotClasses: function iconDotClasses() {
      return this.animate
        ? prefixCls('icon-dot') + ' ' + prefixCls('animate')
        : prefixCls('icon-dot');
    },
    iconText: function iconText() {
      var text;
      switch (this.iconType) {
        case 'input-like':
          text = '赞';
          break;
        case 'input-collect':
          text = '收藏';
          break;
      }
      return text;
    },
  },

  watch: {
    active: function active(newVal, oldVal) {
      if (newVal && newVal != oldVal) {
        this.animate = true;
      } else {
        this.animate = false;
      }
    },
  },

  props: {
    iconType: String,
    count: Number,
    theme: {
      type: String,
      default: 'dp',
    },
    active: {
      type: Boolean,
      default: false,
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.aniBoxClasses},[_c('i',{class:_vm.iconClasses}),_vm._v(" "),_c('i',{class:_vm.iconDotClasses}),_vm._v(" "),(_vm.count > 0 && _vm.count < 10000)?_c('div',{staticClass:"seed-num-box"},[_c('scroll-num',{attrs:{"count":_vm.count,"active":_vm.active}})],1):_vm._e(),_vm._v(" "),(_vm.iconText && _vm.iconText.length !== 0 && _vm.count <= 0)?_c('div',{staticClass:"seed-def-text"},[_vm._v("\n    "+_vm._s(_vm.iconText)+"\n  ")]):_vm._e(),_vm._v(" "),(_vm.count >= 10000)?_c('div',{staticClass:"seed-num-box"},[_vm._v("\n    "+_vm._s(_vm.numberCeil(_vm.count))+"\n  ")]):_vm._e()])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
