// @seed/elements/es/loading/vue/index.js
import { prefixCls, priceZeroClear } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//

var script = {
  name: 'loading',
  // data:function(){
  //     return {

  //     }
  // },
  props: {
    loadingText: {
      type: String,
      default: '',
    },
    loading: {
      type: Boolean,
      default: false,
    },
    vCenter: {
      type: Boolean,
      default: false,
    },
    loadingStyles: {
      type: Object,
    },
  },
  computed: {
    containerClasses: function containerClasses() {
      var obj;

      return [
        prefixCls('loading-container'),
        ( obj = {}, obj[prefixCls('v-center')] = this.vCenter, obj ) ];
    },
    wrapClasses: function wrapClasses() {
      return [prefixCls('load-wrap')];
    },
    imgClasses: function imgClasses() {
      return [prefixCls('loading-img')];
    },
  },
  methods: {
    prefixCls: function prefixCls$1() {
      var arg = [], len = arguments.length;
      while ( len-- ) arg[ len ] = arguments[ len ];

      return prefixCls.apply(void 0, arg);
    },
    priceZeroClear: function priceZeroClear$1() {
      var arg = [], len = arguments.length;
      while ( len-- ) arg[ len ] = arguments[ len ];

      return priceZeroClear.apply(void 0, arg);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.containerClasses,style:(_vm.loadingStyles)},[_c('div',{class:_vm.wrapClasses},[_c('div',{class:_vm.imgClasses}),_vm._v(" "),(_vm.loadingText)?_c('span',[_vm._v(_vm._s(_vm.loadingText))]):_vm._e()])])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
