// @seed/elements/src/checktag/react/index.js
import React, { Component } from 'react';

import PropTypes from 'prop-types';
import cls from 'classnames';
import '../style/index.less';

export default class Checktag extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: props.checked,
    };
    this.onClick = this.onClick.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const { checked, onChange } = nextProps;
    if (nextProps.checked !== this.state.checked) {
      this.setState({
        checked: checked,
      });
      onChange && onChange(checked);
    }
  }

  onClick() {
    const { disabled, onChange } = this.props;
    if (disabled) return;
    const checked = !this.state.checked;
    this.setState({
      checked: checked,
    });
    onChange && onChange(checked);
  }

  render() {
    const { disabled, text, subText } = this.props;
    const isSingleLine = !subText;
    const checktagCls = cls({
      'seed-checktag': true,
      'seed-checktag--single-line': isSingleLine,
      'seed-checktag--multi-line': !isSingleLine,
      'seed-checktag--checked': this.state.checked,
      'seed-checktag--disabled': disabled,
    });
    return (
      <div className={checktagCls} onClick={this.onClick}>
        {isSingleLine ? (
          text
        ) : (
          <div className="seed-checktag__content">
            <p className="seed-checktag__text">{text}</p>
            <p className="seed-checktag__subtext">{subText}</p>
          </div>
        )}
      </div>
    );
  }
}

Checktag.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  text: PropTypes.string,
  subText: PropTypes.string,
  onChange: PropTypes.func,
};
