// @seed/elements/es/currency/vue/index.js
import { prefixCls, priceZeroClear } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

//

var script = {
  name: 'currency',
  data: function() {
    return {
      subTypeText: {
        none: '',
        start: '起',
        avg: '/人',
      },
    };
  },
  props: {
    originPrice: {
      type: Number,
      default: 0,
    },
    price: {
      type: Number,
      default: 0,
    },
    size: {
      type: String,
      default: 's',
      validator: function(value) {
        return ['s', 'm', 'l'].indexOf(value) > -1;
      },
    },
    subType: {
      type: String,
      default: 'none',
    },
    discount: {
      type: String,
      default: '',
    },
    notclearZero: {
      type: Boolean,
      default: false,
    },
    styles: {
      type: Object,
    },
  },
  computed: {
    wrapclasses: function wrapclasses() {
      return [
        prefixCls("price"),
        this.size ? prefixCls(("size_" + (this.size))) : prefixCls('price_m') ];
    },
  },
  methods: {
    prefixCls: function prefixCls$1() {
      var arg = [], len = arguments.length;
      while ( len-- ) arg[ len ] = arguments[ len ];

      return prefixCls.apply(void 0, arg);
    },
    priceZeroClear: function priceZeroClear$1() {
      var arg = [], len = arguments.length;
      while ( len-- ) arg[ len ] = arguments[ len ];

      return priceZeroClear.apply(void 0, arg);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return (_vm.size == 's')?_c('div',{class:_vm.wrapclasses,style:(_vm.styles)},[_vm._v("\n  "+_vm._s(_vm.price ? '￥' + _vm.priceZeroClear(_vm.notclearZero, _vm.price) : '暂无售价')+"\n  "+_vm._s(_vm.subType ? _vm.subTypeText[_vm.subType] : null)+"\n")]):_c('div',{class:_vm.prefixCls('price'),style:(_vm.styles)},[_c('div',{class:_vm.prefixCls(_vm.size ? ("price_" + _vm.size) : 'price_m')},[_c('i',[_vm._v("￥")]),_vm._v(_vm._s(_vm.price ? _vm.priceZeroClear(_vm.notclearZero, _vm.price) : '暂无售价')+"\n  ")]),_vm._v(" "),(_vm.subType && !_vm.originPrice)?_c('span',[_vm._v(_vm._s(_vm.subTypeText[_vm.subType]))]):_vm._e(),_vm._v(" "),_c('div',{class:_vm.prefixCls('origin_price_wrap')},[(_vm.discount)?_c('div',{class:_vm.prefixCls('discount_tag')},[_vm._v("\n      "+_vm._s(_vm.discount)+"\n    ")]):_vm._e(),_vm._v(" "),(_vm.originPrice)?_c('span',[_vm._v("￥"),_c('del',[_vm._v(_vm._s(_vm.priceZeroClear(_vm.notclearZero, _vm.originPrice)))])]):_vm._e()])])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
