// @seed/elements/es/tag/vue/index.js
import { prefixCls } from '@seed/ui-base';
import { _ as __vue_normalize__ } from '../../normalize-component-19a2903f.js';
import '../style/index.css';

/**
 * @since 20181129 15:09
 * @author vivaxy
 */

var SERVICE = 'service';
var RANKING_HORIZONTAL = 'ranking-horizontal';
var RANKING_VERTICAL = 'ranking-vertical';
var CHAIN_CERTIFICATION = 'chain-certification';
var BLACK_PEARL = 'black-pearl';
var MUST_EAT_LIST = 'must-eat-list';
var MUST_PLAY_LIST = 'must-play-list';
var MUST_STROLL_LIST = 'must-stroll-list';
var MUST_LIVE_LIST = 'must-live-list';
var MUST_GO_LIST = 'must-go-list';
var PRIMARY = 'primary';
var DISABLED = 'disabled';
var FLAT = 'flat';
var SECONDARY = 'secondary';
var GHOST = 'ghost';
var FLAT_BLACK = 'flat-black';
var TOPIC = 'topic';
var TOP_QUALITY_REVIEW = 'top-quality-review';
var LEVEL = 'level';
var VIP = 'vip';

var types = /*#__PURE__*/Object.freeze({
  SERVICE: SERVICE,
  RANKING_HORIZONTAL: RANKING_HORIZONTAL,
  RANKING_VERTICAL: RANKING_VERTICAL,
  CHAIN_CERTIFICATION: CHAIN_CERTIFICATION,
  BLACK_PEARL: BLACK_PEARL,
  MUST_EAT_LIST: MUST_EAT_LIST,
  MUST_PLAY_LIST: MUST_PLAY_LIST,
  MUST_STROLL_LIST: MUST_STROLL_LIST,
  MUST_LIVE_LIST: MUST_LIVE_LIST,
  MUST_GO_LIST: MUST_GO_LIST,
  PRIMARY: PRIMARY,
  DISABLED: DISABLED,
  FLAT: FLAT,
  SECONDARY: SECONDARY,
  GHOST: GHOST,
  FLAT_BLACK: FLAT_BLACK,
  TOPIC: TOPIC,
  TOP_QUALITY_REVIEW: TOP_QUALITY_REVIEW,
  LEVEL: LEVEL,
  VIP: VIP
});

/**
 * @since 2018-11-30 10:44
 * @author vivaxy
 */

var BUY = 'buy';
var GROUP_PURCHASE = 'group-purchase';
var CASH_BACK = 'cash-back';
var COUPON = 'coupon';
var PAYMENT = 'payment';
var TICKET = 'ticket';
var REDUCE = 'reduce';
var GIFT = 'gift';
var TOUR = 'tour';
var RENT = 'rent';
var ACTIVITY = 'activity';
var NEW = 'new';
var SALE = 'sale';
var SELL = 'sell';
var BANK_CARD = 'bank-card';
var PRICE = 'price';
var VOUCHER = 'voucher';
var PACKAGE = 'package';
var SMART = 'smart';
var BOOKING = 'booking';
var TICKET2 = 'ticket2';
var HOME = 'home';
var ENDORSEMENT = 'endorsement';
var SEAT = 'seat';
var PHOTO = 'photo';
var CERTIFICATE = 'certificate';
var CARD = 'card';
var NUMBER = 'number';
var COURSE = 'course';
var TAKEAWAY = 'takeaway';
var ORDER = 'order';
var QUEUE = 'queue';
var PIECE = 'piece';
var STYLE = 'style';
var WEDDING_RECEPTION = 'wedding-reception';
var SUNSHINE_RESTAURANT = 'sunshine-restaurant';

var obj;

var serviceTextMap = ( obj = {}, obj[BUY] = null, obj[GROUP_PURCHASE] = '团', obj[CASH_BACK] = '返', obj[COUPON] = '券', obj[PAYMENT] = '付', obj[TICKET] = '票', obj[REDUCE] = '减', obj[GIFT] = '礼', obj[TOUR] = '游', obj[RENT] = '租', obj[ACTIVITY] = '活动', obj[NEW] = '新', obj[SALE] = '促', obj[SELL] = '售', obj[BANK_CARD] = '银', obj[PRICE] = '价', obj[VOUCHER] = '抵', obj[PACKAGE] = '套餐', obj[SMART] = '智', obj[BOOKING] = '订', obj[TICKET2] = '票', obj[HOME] = '家', obj[ENDORSEMENT] = '签', obj[SEAT] = '座', obj[PHOTO] = '照', obj[CERTIFICATE] = '证', obj[CARD] = '卡', obj[NUMBER] = '号', obj[COURSE] = '课', obj[TAKEAWAY] = '外', obj[ORDER] = '点', obj[QUEUE] = '排', obj[PIECE] = '拼', obj[STYLE] = '款', obj[WEDDING_RECEPTION] = '宴', obj[SUNSHINE_RESTAURANT] = null, obj );

/**
 * @since 2018-11-29 15:48
 * @author vivaxy
 */

function toFixed(number, length) {
  if (number >= Math.pow(10, length) || number < 0) {
    throw new Error(("Error number `" + number + "` to fixed length " + length + "."));
  }
  var numberString = String(number);
  return (
    Array.from({ length: length - numberString.length }, function () {
      return 0;
    }) + numberString
  );
}

//

var commonClassName = prefixCls('tag');
var script = {
  name: 'tag',
  data: function data() {
    return { types: types, serviceTextMap: serviceTextMap, toFixed: toFixed, prefixCls: prefixCls };
  },
  computed: {
    classes: function classes() {
      switch (this.type) {
        case SERVICE:
          return {
            service: [
              commonClassName,
              prefixCls('tag', 'service'),
              prefixCls('tag', 'service', this.value) ],
          };
        case RANKING_HORIZONTAL:
          return {
            rankingHorizontal: [
              commonClassName,
              prefixCls('tag', 'ranking-horizontal'),
              prefixCls('tag', 'ranking', this.value < 4 ? 'red' : 'yellow') ],
          };
        case RANKING_VERTICAL:
          return {
            rankingVertical: [
              commonClassName,
              prefixCls('tag', 'ranking-vertical'),
              prefixCls('tag', 'ranking', this.value < 4 ? 'red' : 'yellow') ],
            rankingVerticalTop: prefixCls('tag', 'ranking-vertical-top'),
            rankingVerticalRanking: prefixCls(
              'tag',
              'ranking-vertical-ranking'
            ),
          };
        case CHAIN_CERTIFICATION:
          return {
            chainCertification: [
              commonClassName,
              prefixCls('tag', 'chain-certification') ],
          };
        case BLACK_PEARL:
          return {
            blackPearl: [commonClassName, prefixCls('tag', 'black-pearl')],
          };
        case MUST_EAT_LIST:
          return {
            mustEatList: [commonClassName, prefixCls('tag', 'must-eat-list')],
          };
        case MUST_LIVE_LIST:
          return {
            mustLiveList: [commonClassName, prefixCls('tag', 'must-live-list')],
          };
        case MUST_STROLL_LIST:
          return {
            mustStrollList: [
              commonClassName,
              prefixCls('tag', 'must-stroll-list') ],
          };
        case MUST_PLAY_LIST:
          return {
            mustPlayList: [commonClassName, prefixCls('tag', 'must-play-list')],
          };
        case PRIMARY:
          if (this.size === 'lg') {
            return {
              primary: [
                commonClassName,
                prefixCls('tag', 'primary'),
                prefixCls('tag', 'primary', this.size) ],
            };
          }
          return {
            primary: [
              commonClassName,
              prefixCls('tag', 'primary'),
              prefixCls('tag', 'primary', 'md') ],
          };
        case DISABLED:
          return {
            disabled: [commonClassName, prefixCls('tag', 'disabled')],
          };
        case FLAT:
          return {
            flat: [
              commonClassName,
              prefixCls('tag', 'flat'),
              this.bordered ? prefixCls('tag', 'flat', 'bordered') : '' ],
          };
        case SECONDARY:
          return {
            secondary: [commonClassName, prefixCls('tag', 'secondary')],
          };
        case GHOST:
          return {
            ghost: [commonClassName, prefixCls('tag', 'ghost')],
          };
        case FLAT_BLACK:
          return {
            flatBlack: [commonClassName, prefixCls('tag', 'flat-black')],
          };
        case TOPIC:
          return {
            topic: [commonClassName, prefixCls('tag', 'topic')],
            topicHashTag: prefixCls('tag', 'topic-hash-tag'),
            topicHashTagVertical: prefixCls(
              'tag',
              'topic-hash-tag',
              'vertical'
            ),
            topicHashTagHorizontal: prefixCls(
              'tag',
              'topic-hash-tag',
              'horizontal'
            ),
            topicText: prefixCls('tag', 'topic-text'),
          };
        case TOP_QUALITY_REVIEW:
          return {
            topQualityReview: [
              commonClassName,
              prefixCls('tag', 'top-quality-review'),
              prefixCls('tag', 'top-quality-review', this.size) ],
          };
        case LEVEL:
          return {
            level: [
              commonClassName,
              prefixCls('tag', 'level'),
              prefixCls('tag', 'level', this.value) ],
          };
        case VIP:
          return {
            vip: [
              commonClassName,
              prefixCls('tag', 'vip'),
              prefixCls('tag', 'vip', this.value),
              prefixCls('tag', 'vip', this.size) ],
          };
      }
      return {};
    },
  },
  props: {
    type: {
      type: String,
      validator: function validator(value) {
        return Object.values(types).includes(value);
      },
      required: true,
    },
    value: {
      type: [String, Number],
    },
    size: {
      type: [String, Number],
      validator: function validator(value) {
        return ['lg', 'md', 24, 34, 38, 42].includes(value);
      },
    },
    bordered: {
      type: Boolean,
      default: false,
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',[(_vm.type === _vm.types.SERVICE)?_c('div',{class:_vm.classes.service},[_vm._v("\n    "+_vm._s(_vm.serviceTextMap[_vm.value])+"\n  ")]):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.RANKING_HORIZONTAL)?_c('div',{class:_vm.classes.rankingHorizontal},[_vm._v("\n    TOP "+_vm._s(_vm.value)+"\n  ")]):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.RANKING_VERTICAL)?_c('div',{class:_vm.classes.rankingVertical},[_c('div',{class:_vm.classes.rankingVerticalTop},[_vm._v("TOP")]),_vm._v(" "),_c('div',{class:_vm.classes.rankingVerticalRanking},[_vm._v(_vm._s(_vm.toFixed(_vm.value, 2)))])]):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.CHAIN_CERTIFICATION)?_c('div',{class:_vm.classes.chainCertification},[_vm._v("\n    连锁认证\n  ")]):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.BLACK_PEARL)?_c('div',{class:_vm.classes.blackPearl}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.MUST_EAT_LIST)?_c('div',{class:_vm.classes.mustEatList}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.MUST_LIVE_LIST)?_c('div',{class:_vm.classes.mustLiveList}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.MUST_STROLL_LIST)?_c('div',{class:_vm.classes.mustStrollList}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.MUST_PLAY_LIST)?_c('div',{class:_vm.classes.mustPlayList}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.PRIMARY)?_c('div',{class:_vm.classes.primary},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.DISABLED)?_c('div',{class:_vm.classes.disabled},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.FLAT)?_c('div',{class:_vm.classes.flat},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.SECONDARY)?_c('div',{class:_vm.classes.secondary},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.GHOST)?_c('div',{class:_vm.classes.ghost},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.FLAT_BLACK)?_c('div',{class:_vm.classes.flatBlack},[_vm._t("default")],2):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.TOPIC)?_c('div',{class:_vm.classes.topic},[_c('div',{class:_vm.classes.topicHashTag},[_c('div',{class:_vm.classes.topicHashTagVertical}),_vm._v(" "),_c('div',{class:_vm.classes.topicHashTagHorizontal})]),_vm._v(" "),_c('div',{class:_vm.classes.topicText},[_vm._t("default")],2)]):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.TOP_QUALITY_REVIEW)?_c('div',{class:_vm.classes.topQualityReview}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.LEVEL)?_c('div',{class:_vm.classes.level}):_vm._e(),_vm._v(" "),(_vm.type === _vm.types.VIP)?_c('div',{class:_vm.classes.vip}):_vm._e()])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = __vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

export default index;
