// @dp/rlu2/src/styled-components/response-320.mjs
import { createGlobalStyle } from "styled-components";
const RLUGlobalStyles = createGlobalStyle`
/*reset*/
*{
    margin: 0;
    padding: 0;
}
ul, ol {
    list-style: none;
}
a{
    text-decoration: none;
}
.Hide{
    display: none;
}
.clearfix:before{
    content: " ";
    display: block;
}
.clearfix:after{
    content: " ";
    display: block;
    height: 0;
    clear: both;
}
html {
    font-size: 11.71875px;
}

html {
    font-size: 3.125vw;
}





@media (device-width:320px){

    html {
        font-size: 10px;
    }

}

@media (device-width:360px){

    html {
        font-size: 11.25px;
    }

}

@media (device-width:375px){

    html {
        font-size: 11.71875px;
    }

}

@media (device-width:384px){

    html {
        font-size: 12px;
    }

}

@media (device-width:412px){

    html {
        font-size: 12.8432px;
    }

}

@media (device-width:414px){

    html {
        font-size: 12.875px;
    }

}


@media (device-width:480px){
    html {
        font-size: 15px;
    }
}`;
export default RLUGlobalStyles;
