// @seed/elements/lib/loading/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/defineProperty"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("../style/index.css");

var _classnames = _interopRequireDefault(require("classnames"));

var _uiBase = require("@seed/ui-base");

var Loading =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Loading, _React$Component);

  function Loading(props) {
    (0, _classCallCheck2.default)(this, Loading);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Loading).call(this, props));
  }

  (0, _createClass2.default)(Loading, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          show = _this$props.show,
          loadingText = _this$props.loadingText,
          loadingStyles = _this$props.loadingStyles,
          vCenter = _this$props.vCenter;

      if (show === false) {
        return null;
      }

      return _react.default.createElement("div", {
        style: loadingStyles,
        className: (0, _classnames.default)((0, _uiBase.prefixCls)('loading-container'), (0, _defineProperty2.default)({}, (0, _uiBase.prefixCls)('v-center'), vCenter))
      }, _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)('load-wrap')
      }, _react.default.createElement("div", {
        className: (0, _uiBase.prefixCls)('loading-img')
      }), _react.default.createElement("span", null, loadingText ? loadingText : '')));
    }
  }]);
  return Loading;
}(_react.default.Component);

exports.default = Loading;
Loading.propTypes = {
  loading: _propTypes.default.bool,
  // 增加icon
  // 增加overlay
  vCenter: _propTypes.default.bool,
  loadingStyles: _propTypes.default.object,
  loadingText: _propTypes.default.string
};
