// @seed/elements/lib/tag/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = Tag;

var _values = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/values"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/defineProperty"));

var _parseInt2 = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/parse-int"));

var _concat = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/concat"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _serviceTextMap = _interopRequireDefault(require("../common/service-text-map.js"));

var services = _interopRequireWildcard(require("../common/services.js"));

var types = _interopRequireWildcard(require("../common/types.js"));

var _toFixed = _interopRequireDefault(require("../common/to-fixed.js"));

require("../style/index.css");

var _uiBase = require("@seed/ui-base");

var _definePropTypeByType, _definePropTypeByType2, _definePropTypeByType4;

function Service(_ref) {
  var value = _ref.value;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'service'), (0, _uiBase.prefixCls)('tag', 'service', value))
  }, _serviceTextMap.default[value]);
}

Service.propTypes = {
  value: function value(props, propName, componentName) {
    if (!_serviceTextMap.default.hasOwnProperty(props[propName])) {
      var _context, _context2;

      throw new Error((0, _concat.default)(_context = (0, _concat.default)(_context2 = "Invalid prop `".concat(propName, "` of value `")).call(_context2, props[propName], "` supplied to `")).call(_context, componentName, "`, expected one of `Tag.services`"));
    }
  }
};

function RankingHorizontal(_ref2) {
  var value = _ref2.value;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'ranking-horizontal'), (0, _uiBase.prefixCls)('tag', 'ranking', value < 4 ? 'red' : 'yellow'))
  }, "TOP ", value);
}

RankingHorizontal.propTypes = {
  value: function value(props, propName, componentName) {
    if (typeof props[propName] !== 'number' || (0, _parseInt2.default)(props[propName]) !== props[propName] || props[propName] < 1 || props[propName] > 9) {
      var _context3, _context4;

      throw new Error((0, _concat.default)(_context3 = (0, _concat.default)(_context4 = "Invalid prop `".concat(propName, "` of value `")).call(_context4, props[propName], "` supplied to `")).call(_context3, componentName, "`, expected an integer from 1 to 9."));
    }
  }
};

function RankingVertical(_ref3) {
  var value = _ref3.value;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'ranking-vertical'), (0, _uiBase.prefixCls)('tag', 'ranking', value < 4 ? 'red' : 'yellow'))
  }, _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag', 'ranking-vertical-top'))
  }, "TOP"), _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag', 'ranking-vertical-ranking'))
  }, (0, _toFixed.default)(value, 2)));
}

RankingVertical.propTypes = {
  value: function value(props, propName, componentName) {
    if (typeof props[propName] !== 'number' || (0, _parseInt2.default)(props[propName]) !== props[propName] || props[propName] < 1 || props[propName] > 99) {
      var _context5, _context6;

      throw new Error((0, _concat.default)(_context5 = (0, _concat.default)(_context6 = "Invalid prop `".concat(propName, "` of value `")).call(_context6, props[propName], "` supplied to `")).call(_context5, componentName, "`, expected an integer from 1 to 99."));
    }
  }
};

function ChainCertification() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'chain-certification'))
  }, "\u8FDE\u9501\u8BA4\u8BC1");
}

function BlackPearl() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'black-pearl'))
  });
}

function MustEatList() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'must-eat-list'))
  });
}

function MustLiveList() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'must-live-list'))
  });
}

function MustStrollList() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'must-stroll-list'))
  });
}

function MustPlayList() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'must-play-list'))
  });
}

function MustGoList() {
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'must-go-list'))
  });
}

function Primary(_ref4) {
  var size = _ref4.size,
      children = _ref4.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'primary'), (0, _uiBase.prefixCls)('tag', 'primary', size))
  }, children);
}

Primary.propTypes = {
  size: _propTypes.default.oneOf(['lg', 'md']),
  children: _propTypes.default.node.isRequired
};
Primary.defaultProps = {
  size: 'md'
};

function Disabled(_ref5) {
  var size = _ref5.size,
      children = _ref5.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'disabled'), (0, _uiBase.prefixCls)('tag', 'disabled', size))
  }, children);
}

Disabled.propTypes = {
  size: _propTypes.default.oneOf(['lg', 'md']),
  children: _propTypes.default.node.isRequired
};
Disabled.defaultProps = {
  size: 'md'
};

function Flat(_ref6) {
  var bordered = _ref6.bordered,
      children = _ref6.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'flat'), (0, _defineProperty2.default)({}, (0, _uiBase.prefixCls)('tag', 'flat', 'bordered'), bordered))
  }, children);
}

Flat.propTypes = {
  bordered: _propTypes.default.bool,
  children: _propTypes.default.node.isRequired
};
Flat.defaultProps = {
  bordered: false
};

function Secondary(_ref7) {
  var children = _ref7.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'secondary'))
  }, children);
}

Secondary.propTypes = {
  children: _propTypes.default.node.isRequired
};

function Ghost(_ref8) {
  var children = _ref8.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'ghost'))
  }, children);
}

Ghost.propTypes = {
  children: _propTypes.default.node.isRequired
};

function FlatBlack(_ref9) {
  var children = _ref9.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'flat-black'))
  }, children);
}

FlatBlack.propTypes = {
  children: _propTypes.default.node.isRequired
};

function HashTag() {
  return _react.default.createElement("div", {
    className: (0, _uiBase.prefixCls)('tag', 'topic-hash-tag')
  }, _react.default.createElement("div", {
    className: (0, _uiBase.prefixCls)('tag', 'topic-hash-tag', 'vertical')
  }), _react.default.createElement("div", {
    className: (0, _uiBase.prefixCls)('tag', 'topic-hash-tag', 'horizontal')
  }));
}

function Topic(_ref10) {
  var children = _ref10.children;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'topic'))
  }, _react.default.createElement(HashTag, null), _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag', 'topic-text'))
  }, children));
}

Topic.propTypes = {
  children: _propTypes.default.node.isRequired
};

function TopQualityReview(_ref11) {
  var size = _ref11.size;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'top-quality-review'), (0, _uiBase.prefixCls)('tag', 'top-quality-review', size))
  });
}

TopQualityReview.propType = {
  size: _propTypes.default.oneOf(['lg', 'md'])
};
TopQualityReview.defaultProps = {
  size: 'md'
};

function Level(_ref12) {
  var value = _ref12.value;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'level'), (0, _uiBase.prefixCls)('tag', 'level', value))
  });
}

Level.propTypes = {
  value: function value(props, propName, componentName) {
    if (typeof props[propName] !== 'number' || (0, _parseInt2.default)(props[propName]) !== props[propName] || props[propName] < 1 || props[propName] > 8) {
      var _context7, _context8;

      throw new Error((0, _concat.default)(_context7 = (0, _concat.default)(_context8 = "Invalid prop `".concat(propName, "` of value `")).call(_context8, props[propName], "` supplied to `")).call(_context7, componentName, "`, expected an integer from 1 to 8."));
    }
  }
};

function Vip(_ref13) {
  var value = _ref13.value,
      size = _ref13.size;
  return _react.default.createElement("div", {
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('tag'), (0, _uiBase.prefixCls)('tag', 'vip'), (0, _uiBase.prefixCls)('tag', 'vip', value), (0, _uiBase.prefixCls)('tag', 'vip', size))
  });
}

Vip.propTypes = {
  value: _propTypes.default.oneOf(['red', 'orange', 'blue']).isRequired,
  size: _propTypes.default.oneOf([24, 34, 38, 42]).isRequired
};

function Tag(_ref14) {
  var type = _ref14.type,
      value = _ref14.value,
      size = _ref14.size,
      bordered = _ref14.bordered,
      children = _ref14.children;

  if (type === types.SERVICE) {
    return _react.default.createElement(Service, {
      value: value
    });
  }

  if (type === types.RANKING_HORIZONTAL) {
    return _react.default.createElement(RankingHorizontal, {
      value: value
    });
  }

  if (type === types.RANKING_VERTICAL) {
    return _react.default.createElement(RankingVertical, {
      value: value
    });
  }

  if (type === types.CHAIN_CERTIFICATION) {
    return _react.default.createElement(ChainCertification, null);
  }

  if (type === types.BLACK_PEARL) {
    return _react.default.createElement(BlackPearl, null);
  }

  if (type === types.MUST_EAT_LIST) {
    return _react.default.createElement(MustEatList, null);
  }

  if (type === types.MUST_LIVE_LIST) {
    return _react.default.createElement(MustLiveList, null);
  }

  if (type === types.MUST_STROLL_LIST) {
    return _react.default.createElement(MustStrollList, null);
  }

  if (type === types.MUST_PLAY_LIST) {
    return _react.default.createElement(MustPlayList, null);
  }

  if (type === types.MUST_GO_LIST) {
    return _react.default.createElement(MustGoList, null);
  }

  if (type === types.PRIMARY) {
    return _react.default.createElement(Primary, {
      size: size
    }, children);
  }

  if (type === types.DISABLED) {
    return _react.default.createElement(Disabled, {
      size: size
    }, children);
  }

  if (type === types.FLAT) {
    return _react.default.createElement(Flat, {
      bordered: bordered
    }, children);
  }

  if (type === types.SECONDARY) {
    return _react.default.createElement(Secondary, null, children);
  }

  if (type === types.GHOST) {
    return _react.default.createElement(Ghost, null, children);
  }

  if (type === types.FLAT_BLACK) {
    return _react.default.createElement(FlatBlack, null, children);
  }

  if (type === types.TOPIC) {
    return _react.default.createElement(Topic, null, children);
  }

  if (type === types.TOP_QUALITY_REVIEW) {
    return _react.default.createElement(TopQualityReview, {
      size: size
    });
  }

  if (type === types.LEVEL) {
    return _react.default.createElement(Level, {
      value: value
    });
  }

  if (type === types.VIP) {
    return _react.default.createElement(Vip, {
      value: value,
      size: size
    });
  }
}

Tag.types = types;
Tag.services = services;
Tag.propTypes = {
  type: _propTypes.default.oneOf((0, _values.default)(types)).isRequired,
  value: definePropTypeByType((_definePropTypeByType = {}, (0, _defineProperty2.default)(_definePropTypeByType, types.SERVICE, Service), (0, _defineProperty2.default)(_definePropTypeByType, types.RANKING_HORIZONTAL, RankingHorizontal), (0, _defineProperty2.default)(_definePropTypeByType, types.RANKING_VERTICAL, RankingVertical), (0, _defineProperty2.default)(_definePropTypeByType, types.LEVEL, Level), (0, _defineProperty2.default)(_definePropTypeByType, types.VIP, Vip), _definePropTypeByType)),
  size: definePropTypeByType((_definePropTypeByType2 = {}, (0, _defineProperty2.default)(_definePropTypeByType2, types.PRIMARY, Primary), (0, _defineProperty2.default)(_definePropTypeByType2, types.TOP_QUALITY_REVIEW, TopQualityReview), (0, _defineProperty2.default)(_definePropTypeByType2, types.VIP, Vip), _definePropTypeByType2)),
  bordered: definePropTypeByType((0, _defineProperty2.default)({}, types.FLAT, Flat)),
  children: definePropTypeByType((_definePropTypeByType4 = {}, (0, _defineProperty2.default)(_definePropTypeByType4, types.PRIMARY, Primary), (0, _defineProperty2.default)(_definePropTypeByType4, types.DISABLED, Disabled), (0, _defineProperty2.default)(_definePropTypeByType4, types.SECONDARY, Secondary), (0, _defineProperty2.default)(_definePropTypeByType4, types.GHOST, Ghost), (0, _defineProperty2.default)(_definePropTypeByType4, types.FLAT_BLACK, FlatBlack), (0, _defineProperty2.default)(_definePropTypeByType4, types.TOPIC, Topic), _definePropTypeByType4))
};

function definePropTypeByType(typeComponentMap) {
  var typeKey = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'type';
  return function (props, propName, componentName) {
    if (typeComponentMap.hasOwnProperty(props[typeKey])) {
      return _propTypes.default.checkPropTypes(typeComponentMap[props[typeKey]].propTypes, props, propName, componentName);
    }
  };
}
