// @seed/elements/lib/checktag/vue/index.js
'use strict';

var normalizeComponent = require('../../normalize-component-136d5168.js');
require('../style/index.css');

//

var script = {
  name: 'checktag',
  props: {
    text: {
      type: String,
    },
    subText: {
      type: String,
    },
    checked: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    value: {
      type: Boolean,
      default: false,
    },
  },
  data: function data() {
    return {
      localChecked: false,
    };
  },
  created: function created() {
    this.localChecked = this.value || this.checked;
  },
  watch: {
    value: function value() {
      this.localChecked = this.value;
      this.emitEvent();
    },
    checked: function checked() {
      this.localChecked = this.checked;
      this.emitEvent();
    },
  },
  methods: {
    handleClick: function handleClick() {
      if (this.disabled) { return; }
      this.localChecked = !this.localChecked;
      this.emitEvent();
    },
    emitEvent: function emitEvent() {
      this.$emit('input', this.localChecked);
      this.$emit('change', this.localChecked);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{staticClass:"seed-checktag",class:[
    _vm.subText ? 'seed-checktag--multi-line' : 'seed-checktag--single-line',
    {
      'seed-checktag--checked': _vm.localChecked,
      'seed-checktag--disabled': _vm.disabled,
    } ],on:{"click":_vm.handleClick}},[(_vm.subText)?_c('div',{staticClass:"seed-checktag__content"},[_c('p',{staticClass:"seed-checktag__text"},[_vm._v(_vm._s(_vm.text))]),_vm._v(" "),_c('p',{staticClass:"seed-checktag__subtext"},[_vm._v(_vm._s(_vm.subText))])]):[_vm._v(_vm._s(_vm.text))]],2)};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = normalizeComponent.__vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

module.exports = index;
