// @seed/elements/lib/avatar/react/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = Avatar;

var _assign = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/object/assign"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _index = _interopRequireDefault(require("../../image-cover/react/index"));

require("../style/index.css");

var _classnames = _interopRequireDefault(require("classnames"));

var _uiBase = require("@seed/ui-base");

function Avatar(props) {
  var styles = props.styles,
      width = props.width,
      url = props.url,
      loading = props.loading,
      imageCoverData = props.imageCoverData;
  var wrapStyle = {
    width: width / 2 + 'px',
    height: width / 2 + 'px',
    backgroundImage: url ? "url(".concat(url, ")") : null
  };
  return _react.default.createElement("div", {
    style: (0, _assign.default)({}, wrapStyle, styles),
    className: (0, _classnames.default)((0, _uiBase.prefixCls)('avatar-container'), loading ? (0, _uiBase.prefixCls)('avatar-loading') : '')
  }, imageCoverData && _react.default.createElement(_index.default, imageCoverData));
}

Avatar.propTypes = {
  // 增加icon
  // 增加overlay
  url: _propTypes.default.string,
  styles: _propTypes.default.object,
  width: _propTypes.default.number,
  defaultImg: _propTypes.default.bool,
  imageCoverData: _propTypes.default.object
};
