// @seed/elements/lib/checkbox/vue/index.js
'use strict';

var normalizeComponent = require('../../normalize-component-136d5168.js');
require('../style/index.css');

//
var script = {
  name: 'checkbox',
  props: {
    checked: {
      type: Boolean,
      default: false,
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    value: {
      type: Boolean,
      default: false,
    },
  },
  data: function data() {
    return {
      localChecked: false,
    };
  },
  created: function created() {
    this.localChecked = this.value || this.checked;
  },
  watch: {
    value: function value() {
      this.localChecked = this.value;
      this.emitEvent();
    },
    checked: function checked() {
      this.localChecked = this.checked;
      this.emitEvent();
    },
  },
  methods: {
    handleChange: function handleChange() {
      if (this.disabled) { return; }
      this.localChecked = !this.localChecked;
      this.emitEvent();
    },
    emitEvent: function emitEvent() {
      this.$emit('input', this.localChecked);
      this.$emit('change', this.localChecked);
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('label',{staticClass:"seed-checkbox"},[_c('input',{staticClass:"seed-checkbox__input",attrs:{"type":"checkbox","disabled":_vm.disabled},domProps:{"checked":_vm.localChecked},on:{"change":_vm.handleChange}}),_vm._v(" "),_c('div',{staticClass:"seed-checkbox__mark"})])};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = normalizeComponent.__vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

module.exports = index;
