// @seed/elements/es/separator/react/index.js
import _classCallCheck from "@babel/runtime-corejs3/helpers/classCallCheck";
import _createClass from "@babel/runtime-corejs3/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime-corejs3/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime-corejs3/helpers/getPrototypeOf";
import _inherits from "@babel/runtime-corejs3/helpers/inherits";

/**
 * @since 2018-11-30 17:06
 * @author vivaxy
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cls from 'classnames';
import { prefixCls } from '@seed/ui-base';
import "../style/index.css";

var Separator =
/*#__PURE__*/
function (_Component) {
  _inherits(Separator, _Component);

  function Separator() {
    _classCallCheck(this, Separator);

    return _possibleConstructorReturn(this, _getPrototypeOf(Separator).apply(this, arguments));
  }

  _createClass(Separator, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          size = _this$props.size,
          color = _this$props.color,
          marginHorizontal = _this$props.marginHorizontal;
      return React.createElement("div", {
        className: cls(prefixCls('separator'), prefixCls('separator', 'line', size)),
        style: {
          backgroundColor: color,
          marginLeft: marginHorizontal,
          marginRight: marginHorizontal
        }
      });
    }
  }]);

  return Separator;
}(Component);

Separator.propTypes = {
  size: PropTypes.string,
  color: PropTypes.string,
  marginHorizontal: PropTypes.string
};
Separator.defaultProps = {
  size: 'sm',
  color: null,
  marginHorizontal: '0'
};
export { Separator as default };
