// @seed/elements/src/tag/common/services.js
/**
 * @since 2018-11-30 10:44
 * @author vivaxy
 */

export const BUY = 'buy';
export const GROUP_PURCHASE = 'group-purchase';
export const CASH_BACK = 'cash-back';
export const COUPON = 'coupon';
export const PAYMENT = 'payment';
export const TICKET = 'ticket';
export const REDUCE = 'reduce';
export const GIFT = 'gift';
export const TOUR = 'tour';
export const RENT = 'rent';
export const ACTIVITY = 'activity';
export const NEW = 'new';
export const SALE = 'sale';
export const SELL = 'sell';
export const BANK_CARD = 'bank-card';
export const PRICE = 'price';
export const VOUCHER = 'voucher';
export const PACKAGE = 'package';
export const SMART = 'smart';
export const BOOKING = 'booking';
export const TICKET2 = 'ticket2';
export const HOME = 'home';
export const ENDORSEMENT = 'endorsement';
export const SEAT = 'seat';
export const PHOTO = 'photo';
export const CERTIFICATE = 'certificate';
export const CARD = 'card';
export const NUMBER = 'number';
export const COURSE = 'course';
export const TAKEAWAY = 'takeaway';
export const ORDER = 'order';
export const QUEUE = 'queue';
export const PIECE = 'piece';
export const STYLE = 'style';
export const WEDDING_RECEPTION = 'wedding-reception';
export const SUNSHINE_RESTAURANT = 'sunshine-restaurant';
