// @seed/elements/lib/switch/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/defineProperty"));

var _concat = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/concat"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

var _uiBase = require("@seed/ui-base");

require("../style/index.css");

var Switch =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Switch, _Component);

  function Switch() {
    var _getPrototypeOf2, _context;

    var _this;

    (0, _classCallCheck2.default)(this, Switch);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Switch)).call.apply(_getPrototypeOf2, (0, _concat.default)(_context = [this]).call(_context, args)));

    _this.toggle = function () {
      if (_this.props.disabled) {
        return;
      }

      var checked = !_this.props.checked;

      _this.props.onChange(checked);
    };

    return _this;
  }

  (0, _createClass2.default)(Switch, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          checked = _this$props.checked,
          disabled = _this$props.disabled;
      return _react.default.createElement("span", {
        className: (0, _classnames.default)((0, _uiBase.prefixCls)('switch'), (0, _defineProperty2.default)({}, (0, _uiBase.prefixCls)('switch--checked'), checked), (0, _defineProperty2.default)({}, (0, _uiBase.prefixCls)('switch--disabled'), disabled)),
        onClick: this.toggle
      });
    }
  }]);
  return Switch;
}(_react.Component);

exports.default = Switch;
Switch.propTypes = {
  checked: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  onChange: _propTypes.default.func
};
Switch.defaultProps = {
  checked: false,
  disabled: false,
  onChange: function onChange() {}
};
