// @seed/elements/lib/separator/vue/index.js
'use strict';

var uiBase = require('@seed/ui-base');
var normalizeComponent = require('../../normalize-component-136d5168.js');
require('../style/index.css');

//

var script = {
  name: 'separator',
  props: {
    size: {
      type: String,
      default: 'sm',
      validator: function validator(value) {
        return ['sm', 'lg', 'md'].includes(value);
      },
    },
    color: {
      type: String,
      default: null,
    },
    marginHorizontal: {
      type: String,
      default: '0',
    },
  },
  computed: {
    classes: function classes() {
      return {
        separator: [
          uiBase.prefixCls('separator'),
          uiBase.prefixCls('separator', 'line', this.size) ],
      };
    },
  },
};

/* script */
var __vue_script__ = script;

/* template */
var __vue_render__ = function () {var _vm=this;var _h=_vm.$createElement;var _c=_vm._self._c||_h;return _c('div',{class:_vm.classes.separator,style:('background-color: ' +
      _vm.color +
      '; marginLeft: ' +
      _vm.marginHorizontal +
      '; marginRight: ' +
      _vm.marginHorizontal)})};
var __vue_staticRenderFns__ = [];

  /* style */
  var __vue_inject_styles__ = undefined;
  /* scoped */
  var __vue_scope_id__ = undefined;
  /* module identifier */
  var __vue_module_identifier__ = undefined;
  /* functional template */
  var __vue_is_functional_template__ = false;
  /* style inject */
  
  /* style inject SSR */
  

  
  var index = normalizeComponent.__vue_normalize__(
    { render: __vue_render__, staticRenderFns: __vue_staticRenderFns__ },
    __vue_inject_styles__,
    __vue_script__,
    __vue_scope_id__,
    __vue_is_functional_template__,
    __vue_module_identifier__,
    undefined,
    undefined
  );

module.exports = index;
