// @seed/elements/lib/checktag/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _bind = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/bind"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _classnames = _interopRequireDefault(require("classnames"));

require("../style/index.css");

var Checktag =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Checktag, _Component);

  function Checktag(props) {
    var _context;

    var _this;

    (0, _classCallCheck2.default)(this, Checktag);
    _this = (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Checktag).call(this, props));
    _this.state = {
      checked: props.checked
    };
    _this.onClick = (0, _bind.default)(_context = _this.onClick).call(_context, (0, _assertThisInitialized2.default)(_this));
    return _this;
  }

  (0, _createClass2.default)(Checktag, [{
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      var checked = nextProps.checked,
          onChange = nextProps.onChange;

      if (nextProps.checked !== this.state.checked) {
        this.setState({
          checked: checked
        });
        onChange && onChange(checked);
      }
    }
  }, {
    key: "onClick",
    value: function onClick() {
      var _this$props = this.props,
          disabled = _this$props.disabled,
          onChange = _this$props.onChange;
      if (disabled) return;
      var checked = !this.state.checked;
      this.setState({
        checked: checked
      });
      onChange && onChange(checked);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props2 = this.props,
          disabled = _this$props2.disabled,
          text = _this$props2.text,
          subText = _this$props2.subText;
      var isSingleLine = !subText;
      var checktagCls = (0, _classnames.default)({
        'seed-checktag': true,
        'seed-checktag--single-line': isSingleLine,
        'seed-checktag--multi-line': !isSingleLine,
        'seed-checktag--checked': this.state.checked,
        'seed-checktag--disabled': disabled
      });
      return _react.default.createElement("div", {
        className: checktagCls,
        onClick: this.onClick
      }, isSingleLine ? text : _react.default.createElement("div", {
        className: "seed-checktag__content"
      }, _react.default.createElement("p", {
        className: "seed-checktag__text"
      }, text), _react.default.createElement("p", {
        className: "seed-checktag__subtext"
      }, subText)));
    }
  }]);
  return Checktag;
}(_react.Component);

exports.default = Checktag;
Checktag.propTypes = {
  checked: _propTypes.default.bool,
  disabled: _propTypes.default.bool,
  text: _propTypes.default.string,
  subText: _propTypes.default.string,
  onChange: _propTypes.default.func
};
