// @seed/elements/src/checkbox/react/index.js
import React, { Component } from 'react';

import PropTypes from 'prop-types';
import '../style/index.less';

export default class Checkbox extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: props.checked,
    };
    this.onChange = this.onChange.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    const { checked, onChange } = nextProps;
    if (nextProps.checked !== this.state.checked) {
      this.setState({
        checked: checked,
      });
      onChange && onChange(checked);
    }
  }

  onChange() {
    const { disabled, onChange } = this.props;
    if (disabled) return;
    const checked = !this.state.checked;
    this.setState({
      checked: checked,
    });
    onChange && onChange(checked);
  }

  render() {
    const { disabled } = this.props;
    return (
      <label className="seed-checkbox">
        <input
          type="checkbox"
          className="seed-checkbox__input"
          checked={this.state.checked}
          disabled={disabled}
          onChange={this.onChange}
        />
        <div className="seed-checkbox__mark"></div>
      </label>
    );
  }
}

Checkbox.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};
