// @seed/elements/es/image-cover/react/index.js
import _Object$values from "@babel/runtime-corejs3/core-js-stable/object/values";
import _indexOfInstanceProperty from "@babel/runtime-corejs3/core-js-stable/instance/index-of";
import React from 'react';
import PropTypes from 'prop-types';
import "../style/index.css";
export var PRIORITY_ENUM = {
  HIGH: 'HIGH',
  LOW: 'LOW'
};
export default function ImageCover(props) {
  var url = props.url,
      className = props.className,
      priority = props.priority;
  var displayUrl = url;

  if (url && _indexOfInstanceProperty(url).call(url, 'http') < 0) {
    displayUrl = "https:".concat(url);
  }

  return React.createElement("div", {
    className: "seed-img-cover-wrap ".concat(className)
  }, priority === PRIORITY_ENUM.HIGH && React.createElement("img", {
    className: "seed-img-src",
    src: displayUrl,
    alt: ""
  }), priority === PRIORITY_ENUM.LOW && React.createElement("div", {
    className: "seed-img-bg",
    style: {
      backgroundImage: "url(".concat(displayUrl, ")")
    }
  }));
}
ImageCover.propTypes = {
  url: PropTypes.string,
  priority: PropTypes.oneOf(_Object$values(PRIORITY_ENUM)),
  className: PropTypes.string
};
ImageCover.defaultProps = {
  url: '',
  priority: PRIORITY_ENUM.HIGH,
  className: ''
};
