// @seed/elements/lib/badge/react/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime-corejs3/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/inherits"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

require("../style/index.css");

var Badge =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Badge, _Component);

  function Badge() {
    (0, _classCallCheck2.default)(this, Badge);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Badge).apply(this, arguments));
  }

  (0, _createClass2.default)(Badge, [{
    key: "renderRedDot",
    value: function renderRedDot() {
      return _react.default.createElement("div", {
        className: "seed-reddot"
      });
    }
  }, {
    key: "renderBadge",
    value: function renderBadge() {
      var children = this.props.children;
      return _react.default.createElement("div", {
        className: "seed-badge"
      }, children);
    }
  }, {
    key: "render",
    value: function render() {
      var type = this.props.type;
      return type === 'normal' ? this.renderBadge() : this.renderRedDot();
    }
  }]);
  return Badge;
}(_react.Component);

exports.default = Badge;
Badge.propTypes = {
  type: _propTypes.default.oneOf(['dot', 'normal']),
  children: _propTypes.default.string
};
